### ISSUE TITLE
Any guidance on how to use a custom language definition?

### ISSUE THREAD

**Issue by shepmaster** [bvaughn/react-highlight.js#5]
I'd like to highlight some custom output, so I'm going through the process of creating a custom language. Any idea how I would configure react-highlight.js to know of the existence of that language?

**Comment by bvaughn** [comment]
Hey @shepmaster,

Not sure I understand. You're adding a custom language definition to Highlight.js? I think that's more a question for that library than `react-highlight`. This is just a convenience wrapper. It should pass through your custom language id/string without any problems, but I don't know how you'd go about registering the definition(s). I haven't done that before. :)

**Comment by shepmaster** [comment]
Ah, sorry to be confusing. Highlight.js has pretty good documentation about how to do it. For example, in my application entrypoint, I have something like this:

```js
import hljs from 'highlight.js';

// Disable all the auto detection
hljs.configure({languages: []});

// And register my language
hljs.registerLanguage('errors', (hljs) => ({
  case_insensitive: true,
  keywords: 'error'
  // All the other interesting stuff
}));
```

The problem is that the value returned from `import hljs from 'highlight.js';` is unique each time it is called; there's no global value. That's fine, except for the fact that this component *also* does `import highlight from 'highlight.js'`, which I don't have any way of accessing.

Some way of getting access to the HLjs object that the component creates or of providing the HLjs object to the component would allow extra configuration to occur. It might also allow for some amount of memory saving; if there are many `<Highlight>` components that each have their own copy of HLjs, that might be bloating up things.

This is all based on my quick investigation, so it's possible I'm missing something.

**Comment by bvaughn** [comment]
Thanks for the additional context. This makes more sense now. :)

Seems like exposing the inner instance of `highlight` would be reasonable. Then your code could do something like...

```
<Highlight
  language={language}
  ref={(ref) => this._highlight = ref.highlight}
>
  {content}
</Highlight>
```

**Comment by shepmaster** [comment]
Actually, I might be wrong in my diagnosis. I think that the React component just initializes HLjs first, before I have a chance to set up anything. Perhaps I can just rearrange some code. Lemme try poking a bit at that before you make any changes; I don't want to waste your time!

**Comment by shepmaster** [comment]
Oh. Oh! This component [embeds Highlight.js](https://github.com/bvaughn/react-highlight.js/blob/681be767dc7cd98178ba9979d059aa81682eb733/dist/main.js#L3290)! That's why there are two objects - there are two libraries! 

Maybe I'm doing something dumb with importing? I ran `npm install --save react-highlight.js` which also brought in `highlight.js`. Is there some way for me to point at the source and not the dist?

**Comment by bvaughn** [comment]
D'oh! This is a problem with my webpack config. I need to add an entry to [`externals`](https://github.com/bvaughn/react-highlight.js/blob/master/webpack.config.dist.js#L14) to avoid inlining Highlight.js

**Comment by shepmaster** [comment]
That's great! I'm glad you understand the problem. I'm just waving my arms around wildly to make people think I know what I'm doing when it comes to JS stuff ;-)

**Comment by bvaughn** [comment]
Haha, no worries. I'll try to get an update pushed for you to try within an hour or two. In meetings in the meanwhile.

**Comment by bvaughn** [comment]
@shepmaster Try version 1.0.4

**Comment by shepmaster** [comment]
Yes, that all works fine now. My original code works as-is. Since it's run before the React component is ever mounted, all my configuration takes place before the first highlighting.

Thanks a bunch! 🌴 ❤️ 💙 💚 🌴

**Comment by shepmaster** [comment]
Closed via a81b8f86bb63d3fff811bf0a0c73e91013c2c5da.

**Comment by bvaughn** [comment]
Huzzah! Glad to hear. :)
